tests/test_chiral_restraints.py

```diff
@@ -0,0 +1,526 @@
+
+import jax
+import jax.numpy as jnp
+import numpy as np
+import scipy
+from rdkit import Chem
+
+from timemachine.fe import chiral_utils, topology, utils
+from timemachine.ff import Forcefield
+from timemachine.integrator import simulate
+from timemachine.potentials.chiral_restraints import (
+    U_chiral_atom,
+    U_chiral_atom_batch,
+    U_chiral_bond,
+    U_chiral_bond_batch,
+    pyramidal_volume,
+    torsion_volume,
+)
+
+
+def minimize_scipy(x0, U_fn):
+    N = x0.shape[0]
+
+    def U_bfgs(x_flat):
+        x_full = x_flat.reshape(N, 3)
+        return U_fn(x_full)
+
+    grad_bfgs_fn = jax.grad(U_bfgs)
+    res = scipy.optimize.minimize(U_bfgs, x0.reshape(-1), jac=grad_bfgs_fn)
+    xi = res.x.reshape(N, 3)
+    return xi
+
+
+def simulate_system(U_fn, x0, num_samples=20000):
+    num_atoms = x0.shape[0]
+    x_min = minimize_scipy(x0, U_fn)
+    seed = 2023
+
+    num_workers = multiprocessing.cpu_count()
+    samples_per_worker = int(np.ceil(num_samples / num_workers))
+
+    # batches_per_worker = num_samples // num_workers
+    burn_in_batches = 2000
+    frames, _ = simulate(
+        x_min, U_fn, 300.0, np.ones(num_atoms) * 4.0, 500, samples_per_worker + burn_in_batches, num_workers, seed=seed
+    )
+    # (ytz): discard burn in batches
+    frames = frames[:, burn_in_batches:, :, :]
+    # collect over all workers
+    frames = frames.reshape(-1, num_atoms, 3)[:num_samples]
+    # sanity check that we didn't undersample
+    assert len(frames) == num_samples
+    return frames
+
+
+def test_setup_chiral_atom_restraints():
+    mol = Chem.MolFromMolBlock(
+        """
+  Mrv2202 06072215563D
+
+  5  4  0  0  0  0            999 V2000
+    0.0000    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633   -0.5138    0.8900 H   0  0  0  0  0  0  0  0  0  0  0  0
+    1.0900    0.0000    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633    1.0277    0.0000 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3633   -0.5138   -0.8900 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0  0  0  0
+  1  3  1  0  0  0  0
+  1  4  1  0  0  0  0
+  1  5  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    # needs to be batched in order for jax to play nicely
+    x0 = utils.get_romol_conf(mol)
+    normal_restr_idxs = chiral_utils.setup_chiral_atom_restraints(mol, x0, 0)
+
+    x0_inverted = [x0[0], x0[2], x0[1], x0[3], x0[4]]
+
+    inverted_restr_idxs = chiral_utils.setup_chiral_atom_restraints(mol, x0_inverted, 0)
+
+    # check the sign of the resulting idxs
+    k = 1000.0
+    assert np.all(np.asarray(U_chiral_atom_batch(x0, normal_restr_idxs, k)) == 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0, inverted_restr_idxs, k)) > 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0_inverted, normal_restr_idxs, k)) > 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0_inverted, inverted_restr_idxs, k)) == 0)
+
+
+def test_setup_chiral_bond_restraints():
```

Comment:
Add one-sentence description of what is being asserted
```suggestion
def test_setup_chiral_bond_restraints():
    """On a 'Cl/C(F)=N/F' conformer, assert that flipping a dihedral angle or permuting restr_idxs
    both independently toggle the chiral bond restraint"""
```